saiph/reduction/utils/svd_test.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+import pandas as pd
+import pytest
+from numpy.testing import assert_array_almost_equal
+from numpy.typing import NDArray
+
+from saiph.reduction.utils.svd import (
+    get_direct_randomized_svd,
+    get_randomized_subspace_iteration,
+    get_svd,
+)
+
+
+# Matrix to decompose
+@pytest.fixture
+def test_matrix() -> pd.DataFrame:
+    A: NDArray[np.float_] = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+    A = pd.DataFrame(A)
+    return A
+
+
+def test_full_svd(test_matrix: pd.DataFrame) -> None:
+    expected_U: NDArray[np.float_] = np.array(
+        [
+            [0.21483724, 0.88723069, -0.40824829],
+            [0.52058739, 0.24964395, 0.81649658],
+            [0.82633754, -0.38794278, -0.40824829],
+        ]
+    )
```

Comment:
how do you compute your `expected`s ? did you check them by hand ? are you sure they are correct ?

Reply:
@jpetot computed using `scipy`.
The reconstruction `U @ np.diag(S) @ Vt` gives the original matrix for both randomized and full SVD (up to the precision of a `float64`)